lang/libcxx/Portfile

```diff
@@ -15,38 +14,103 @@ long_description        ${description} \
                         replace the host version in order to be used.  On Snow Leopard and earlier, this is done \
                         automatically because there is no existing host version of this library.  On Lion and later, \
                         users must build the port with +replacemnt_libcxx and install the root manually if they wish \
-                        to replace the existing host implementation.
+                        to replace the existing host implementation. They can also use the +runtime variant which installs \
+                        copies of the library that can be loaded via DYLD_INSERT_LIBRARIES or the optlibs script\; \
+                        applications started that way will then use a more recent libc++ than the host provides.
 
 homepage                http://libcxx.llvm.org/
 
 master_sites            http://www.llvm.org/releases/${version}/
 dist_subdir             llvm
 
 use_xz                  yes
+set use_cmake           yes
 
 set libcxxabi_distname  libcxxabi-${version}.src
 set libcxx_distname     libcxx-${version}.src
 
 distfiles               ${libcxxabi_distname}${extract.suffix} \
                         ${libcxx_distname}${extract.suffix}
 
-checksums               libcxxabi-5.0.1.src.tar.xz \
-                        rmd160  93b74a0644c1afd565938fba5e21b200b206c72a \
-                        sha256  5a25152cb7f21e3c223ad36a1022faeb8a5ac27c9e75936a5ae2d3ac48f6e854 \
-                        libcxx-5.0.1.src.tar.xz \
-                        rmd160  1e5a22e62515382488f232039e31f0b12c08ca4f \
-                        sha256  fa8f99dd2bde109daa3276d529851a3bce5718d46ce1c5d0806f46caa3e57c00
+checksums               ${libcxxabi_distname}${extract.suffix} \
+                        rmd160  93f0cf438a5168e225e11c23a0cda6695fe3b518 \
+                        sha256  91c6d9c5426306ce28d0627d6a4448e7d164d6a3f64b01cb1d196003b16d641b \
+                        ${libcxx_distname}${extract.suffix} \
+                        rmd160  5d311ab0546181cf3084c520826786a7a9681a95 \
+                        sha256  70931a87bde9d358af6cb7869e7535ec6b015f7e6df64def6d2ecdd954040dd9
+
+if {[tbool use_cmake]} {
+    revision            [expr ${revision} + 1]
+    PortGroup           cmake 1.1
```

Comment:
cmake requires libcxx; therefore libcxx cannot require cmake, because that would be a circular dependency and MacPorts doesn't support that and its behavior in that case is undefined. See https://trac.macports.org/ticket/56519.

Reply:
CMake has started requiring C++11 only relatively recently, and doesn't require libcxx for that on 10.8 and earlier only.

Wouldn't it make sense to provide a pre-C++11 version of cmake for this kind of situation (via [sub]port or variant) provided that version is new enough to build libc++)? We're only talking about a build dependency here (= bootstrap situation), and the CMake version used for building a port should not affect the end result  (= no reproducible build principle issues).